train_gpt2.cu

```diff
@@ -1183,9 +1175,11 @@ void save_state(const char* filename, int step, GPT2* model, DataLoader* loader)
     state_header[10] = step; // step of the optimization
     // model state, state, start at 20 to leave some padding
     *((unsigned long long*)&state_header[20]) = model->rng_state; // random number generator state
-    // dataloader state, start at 30 to leave some padding
+    // dataloader state, start at 30 to leave some padding. Testing currently doesn't have a dataloader
+#ifndef TESTING
```

Comment:
should we instead check if `loader` is `nullptr`?

Reply:
to me, that would imply that `loader` being `nullptr` is part of the normal operating conditions of this function, whereas the ifdef makes it look more like what it truly is, a hack to get some determinism checks working